gems/aws-sdk-core/lib/aws-sdk-core/client_stubs.rb

```diff
@@ -17,6 +20,15 @@ def setup_stubbing
           apply_stubs(operation_name, Array === stubs ? stubs : [stubs])
         end
       end
+
+      # When a client is stubbed allow the user to access the requests made
+      @api_requests = []
+
+      requests = @api_requests
+      self.handle do |context|
```

Comment:
I think as discussed before, we should have a more complex object here, so long as we're actually providing this functionality as a first class member. Maybe create a dedicated class for this, even potentially a dedicated handler class which is inserted when we stub the client.

But for a dedicated class, we could also record more data about each request. We can record the parameters, keep the context object in full, record the method name, and provide convenient helper methods for each.

Reply:
I'd +1 for this, any follow up or discussion on the advice?

Reply:
There was some brief follow up in the outdated comments after Alex cross checked against aws-record